### DISCUSSION TITLE
Search for IP ranges

### DISCUSSION THREAD

**Discussion by randomstr1ng** [Hunter-How/Support#7] [Q&A]
Many Business Applications like SAP generate there ports dynamically based on e.g. Instance numbers or other indicators. This causes that the port a service is listen on varies within a range of (in case of SAP) 100 ports

Is it possible to search for e.g. `ip.port="3200-3299"`?

**Comment by HowHunterHow** [comment]
The best solution for your scenario could be generating a specific fingerprint for this application. It improves your query accuracy and shows the distribution of ports easily.

**Comment by randomstr1ng** [comment]
@HowHunterHow thank you for your response. I aggree that it would make sense to work with a specific fingerprint.
But to avoid false positives, the ability to restrict the search on a range of ports common for a service would be a massive improvement.

Speaking for SAP, the service port will not repsond with anything as long as correctly formed package is send. A collection of nmap fingerprint examples can be found [here](https://github.com/gelim/nmap-erpscan)
With those the Service could be identified also by fingerprints in the search.

Any chance to get this via the search?

**Reply by HowHunterHow** [reply]
@randomstr1ng Our product manager will take your suggestions about the port range. 
When you look for HTTP or HTTPStps services, we suggest you refer to the link you have sent us.
For example,
SAP Start Service: (header.server="gSOAP" && protocol.banner="SID"&& protocol.banner="HOST") || (header.server="gSOAP" && protocol.banner="Location"&& protocol.banner="sapmc")
SAP HANA XS: web.title="SAP XSEngine"

For NO-WEB Services, we will refer to this repository to improve protocol recognition capabilities ASAP. 

YOUR SUGGESTIONS MAKE US BETTER! WE ENCOURAGE ALL OUR USERS TO SPEAK OUT THEIR THOUGHTS.

**Comment by randomstr1ng** [comment]
@HowHunterHow thank you for your response and I really appreciate you are looking into supporting IP ranges and updating protocol detections!

Can you please keep me updated as I'm the leader of the OWASP [SAP Atack Surface Discovery Project](https://github.com/SecuritySilverbacks/SAP-AttackSurfaceDiscovery) and I would love to include hunter as a resource as shodan is very limited!

Thank you!

**Reply by HowHunterHow** [reply]
@randomstr1ng We encourage our users to integrate our services into their open-source projects.  I will update you after we fix this issue.

**Comment by HowHunterHow** [comment]
@randomstr1ng We have managed to detect a few SAP products. You could try out these.
<img width="556" alt="Screen Shot 2023-12-25 at 11 26 15" src="https://github.com/Hunter-How/Support/assets/112148057/053c6def-263a-4c5c-9304-32659cf14451">
For the other protocols such as sapjms, we do not have a corresponding target test environment. If you happen to have any test environment please give us some clues.

**Reply by randomstr1ng** [reply]
@HowHunterHow this looks awesome! Thank you!

regarding a test environment:
- you can either use this https://github.com/randomstr1ng/tf-sap-automation which allows to setup some SAP Components
- or feel free to send me a message in e.g. Discord (same nickname) and I can provide you with access to my private environment ( I try to avoid that all services get publshed external)

**Reply by HowHunterHow** [reply]
@randomstr1ng We have faced some issues setting up some SAP Components. I have added you through Discord (wnck_aeric).
